app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,82 @@
+class OrderItemsController < ApplicationController
+  before_action :check_if_product_is_in_cart, only: [:create]
+
+  def create
+    @order = current_order
+    @order_item = @order.order_items.create(order_item_params)
+    if @order_item.save
+      redirect_to cart_path
+    else
+      if @order_item.product.inventory == 1
+        flash[:error] = "Sorry, there is only #{@order_item.product.inventory} #{@order_item.product.name} available."
+      elsif @order_item.product.inventory == 0
+        flash[:error] = "Sorry, there are no #{@order_item.product.name.pluralize} available."
+      else
+        flash[:error] = "Sorry, there are only #{@order_item.product.inventory} #{@order_item.product.name.pluralize} available."
+      end
+      redirect_to product_path(@order_item.product)
+    end
+  end
+
+  def destroy
+    @order = current_order
+    @order_item = @order.order_items.find(params[:id])
+    @order_item.destroy
+    @order_items = @order.order_items
+
+    redirect_to cart_path
+  end
+
+  def update
+    @order_item = OrderItem.find(params[:id])
+    @order_item.update_attributes(order_item_params)
+    if @order_item.save
+      redirect_to cart_path
+    else
+      flash[:error] = "Sorry, there are only #{@order_item.product.inventory} #{@order_item.product.name.pluralize} available."
+      redirect_to cart_path
+    end
+  end
+
+  def check_if_product_is_in_cart
+    order = current_order
+    @product_id = params[:order_item][:product_id]
+    product_in_cart = order.order_items.select {|order_item| order_item.product_id == @product_id.to_i}
+
+    if !product_in_cart.empty?
+      update_product_in_cart(product_in_cart[0])
+    end
+  end
+
+  def update_product_in_cart(product_in_cart)
+    current_quantity = product_in_cart.quantity
+    additional_quantity = params[:order_item][:quantity].to_i
```

Comment:
It's fairly minor, but I think this method would be better if we wrote it to accept two parameters (the second being the additional quantity).

Doing this would allow for a small improvement in the "Single Responsibility Principle" that Sandi Metz talks about. Right now the method has to know (1) how to set a new value for an existing order item's quantity and (2) how to determine the additional quantity value to add in part (1).
If we provide the quantity directly, then the method only has to deal with the first responsibility, and we can move the code that knows the quantity comes from `params` somewhere else.